src/integrity-checker.js

```diff
@@ -33,30 +39,37 @@ export default class InstallationIntegrityChecker {
    * write a new one.
    */
 
-  async _getIntegrityHashLocation(): Promise<string> {
+  async _getIntegrityHashLocation(usedRegistries?: Set<RegistryNames>): Promise<IntegrityHashLocation> {
     // build up possible folders
+    let registries = registryNames;
+    if (usedRegistries && usedRegistries.size > 0) {
+      registries = usedRegistries;
+    }
```

Comment:
Even if it's currently the case here, you might want to put a comment to explain that passing falsy values for `usedRegistries` is only acceptable when one only needs to read the file but not write to it - otherwise multiple integrity files might get generated!